farm-staking/farm-staking-proxy/src/lib.rs

```diff
@@ -30,52 +30,31 @@ pub trait FarmStakingProxy:
         staking_farm_token_id: TokenIdentifier,
         lp_token_id: TokenIdentifier,
     ) {
-        require!(
-            self.blockchain().is_smart_contract(&lp_farm_address),
-            "Invalid LP Farm address"
-        );
-        require!(
-            self.blockchain().is_smart_contract(&staking_farm_address),
-            "Invalid Staking Farm address"
-        );
-        require!(
-            self.blockchain().is_smart_contract(&pair_address),
-            "Invalid Pair address"
-        );
-        require!(
-            staking_token_id.is_valid_esdt_identifier(),
-            "Invalid Staking token ID"
-        );
-        require!(
-            lp_farm_token_id.is_valid_esdt_identifier(),
-            "Invalid Farm token ID"
-        );
-        require!(
-            staking_farm_token_id.is_valid_esdt_identifier(),
-            "Invalid Staking Farm token ID"
-        );
+        self.require_sc_address(&lp_farm_address);
+        self.require_sc_address(&staking_farm_address);
+        self.require_sc_address(&pair_address);
 
-        require!(
-            lp_token_id.is_valid_esdt_identifier(),
-            "Invalide LP token ID"
-        );
+        self.require_valid_token_id(&staking_token_id);
+        self.require_valid_token_id(&lp_farm_token_id);
+        self.require_valid_token_id(&staking_farm_token_id);
+        self.require_valid_token_id(&lp_token_id);
 
         self.lp_farm_address().set(&lp_farm_address);
```

Comment:
Why not `set_if_empty` here as well?

Reply:
Thought it might be a good idea to let those be overwritten, but it's probably best with set_if_empty here as well.

Reply:
Done.